kv-store/src/rocksdb_kv/mod.rs

```diff
@@ -0,0 +1,190 @@
+use super::*;
+use mktemp::Temp;
+use rocksdb::{checkpoint, DB};
+
+pub struct RocksDB {
+    db: DB,
+    checkpoint: Option<Temp>,
+}
+
+#[async_trait]
+impl KVStore for RocksDB {
+    async fn new(path: &str) -> Result<Self, ()>
+    where
+        Self: Sized,
+    {
+        Ok(RocksDB {
+            db: DB::open_default(path).unwrap(),
+            checkpoint: None,
+        })
+    }
+    async fn open(path: &str) -> Result<Self, ()>
```

Comment:
What's the difference between `fn new` and `fn open`?

Reply:
In rocksdb, logically there is no difference, because rocksdb takes both new / open as the same case and can cover both by open function.

By the way, generally, new is for empty new db, open is for existing db.